session/session.go

```diff
@@ -670,6 +670,15 @@ func (s *session) retry(ctx context.Context, maxCnt uint) (err error) {
 					s.sessionVars.TxnCtx.IsPessimistic = true
 				}
 			}
+			// Complete the missing settings using `NewTxn` compared with lazy activation `txn.Txn(true)`.
+			// More refactors are needed in the future to make the inner logic of transaction activation
+			// interfaces the same.
+			if s.sessionVars.TxnCtx.IsPessimistic {
```

Comment:
I think pessimistic transactions could not retry.

Reply:
It's used to decide the retrying mode using optimistic or pessimistic mode, the retryable condition is not affected by this. Or do you mean if the pessimsitic mode retrying has failed, the retry should abort?